### DISCUSSION TITLE
wrong redirect when using a $queryString while serving from a subfolder

### DISCUSSION THREAD

**Discussion by felix-exon** [livewire/livewire#4842] [7. Bugs (archived)]
I have an app served in a subfolder: `https://mydomain.com/subfolder`.
In the `config/livewire.php` I've set `'asset_url' => /subfolder` which works without any problem - all livewire scripts/styles load and resolve correctly.

this is part of my `.env`:
```env
APP_URL=https://mydomain.com/subfolder
ASSET_URL="/subfolder"
```
I've managed to reproduce with a minimal example.

1. Create a new component `php artisan make:livewire dummy-component`.
2. Leave the file `/resources/livewire/dummy-component.blade.php` as it is
3. Change `app/Http/Livewire/DummyComponent` to this:

```php
<?php

namespace App\Http\Livewire;

use Livewire\Component;

class DummyComponent extends Component
{
    protected $queryString = [
        'selectedName' => [ 'except' => '']
    ];

    public string $selectedName = "";

    public function render()
    {
        return view('livewire.dummy-component')
        ->extends('layouts.layout');
    }
}
```

4. Set up a route like this: `Route::get('/dummy', \App\Http\Livewire\DummyComponent::class);` in `routes/web.php`.
5. Visit `https:<your-root>/subfolder/dummy`

❌ it will redirect to `https:<your-root>/subfolder/subfolder/dummy` 

Evaluating the source of `https:<your-root>/subfolder/dummy`  I've found out that the path is wrong for the `wire:initial-data`:

```html
<div wire:id="M4AD4nOUjRQ1fr3v8GO7" 
        wire:initial-data="{
  "fingerprint": {
    "id": "M4AD4nOUjRQ1fr3v8GO7",
    "name": "dummycomponent",
    "locale": "de",
    "path": "subfolder\\/dummy", ✅  
    "method": "GET",
    "v": "acj"
  },
  "effects": {
    "listeners": [],
    "path": "https:\\/\\/mydomain.com\\/subfolder\\/subfolder\\/dummy" ❌  
  },
  "serverMemo": {
    "children": [],
    "errors": [],
    "htmlHash": "5cd7e88b",
    "data": {
      "selectedName": ""
    },
    "dataMeta": [],
    "checksum": "80d8cf7b6b536c4d6a7c447c3ede5b5ca7fd127c3f035879a983e37f4fab7184"
  }
}">
</div>

<!-- Livewire Component wire-end:M4AD4nOUjRQ1fr3v8GO7 -->
```

To me it seems that this piece of code is not applying the setting made in `config/livewire.php`.

**Comment by joshhanley** [comment]
@deibertf thanks for reporting! Any chance you could create a small example repo which reproduces the issue with as little code as possible so we can investigate it?
